assets/src/stories-editor/index.js

```diff
@@ -269,6 +275,28 @@ store.subscribe( () => {
 			} );
 		}
 	}
+
+	/*
+	 * Prevent an issue where duplicated page blocks had the same anchor attribute, and didn't look right on the front-end.
+	 * @see https://github.com/ampproject/amp-wp/issues/2510
+	 */
+	const usedAnchors = [];
+	const conditionallyUpdateAnchor = ( block ) => {
+		if ( ! has( block, [ 'attributes', 'anchor' ] ) ) {
+			return;
+		}
+
+		if ( usedAnchors.includes( block.attributes.anchor ) ) {
+			updateBlockAttributes( block.clientId, { anchor: block.clientId } );
```

Comment:
This doesn't work when `block.attributes.anchor === block.clientId`. Probably easier to just do something like this:

```js
if ( block.attributes.anchor !=== block.clientId ) {
  updateBlockAttributes( block.clientId, { anchor: block.clientId } );
}
```

Reply:
Sure, good idea. https://github.com/ampproject/amp-wp/pull/2540/commits/ba4f1558a650eb29a1046645074d820e0eb3a1e8 uses your snippet above.